test/suites/helpers.js

```diff
@@ -223,6 +223,26 @@ describe('Helpers', function() {
         [{name: 'John'}, {name: 'Paul'}]
       );
     });
+
+    describe('Issue #472 - tree/cursor.splice does not conform with the specification as of ES6 (ECMAScript 2015)', function () {
+      it('should be possible to splice an array when omitting the nb (deleteCount) argument', function () {
+        const array = [0, 1, 2, 3, 4];
+
+        assert.deepEqual(splice(array, 2), [0, 1]);
+
+        assert.deepEqual(splice(array, -2), [0, 1, 2]);
+
+        assert.deepEqual(splice(array, 2, null), [0, 1, 2, 3, 4]);
+      });
+
+      it('should delete no values when supplying an argument which is not parseable as number', function () {
```

Comment:
Personally, I also tend to think that failing fast is better. But since it means more tests & documentation, I'll leave the decision to you. If you want to edit the PR, I'll wait, else, I'll merge and release since we can also do this modification in the future.